sys/phydat/phydat.c

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2018 Eistec AB
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_phydat
+ * @{
+ *
+ * @file
+ * @brief       Generic sensor/actuator data handling
+ *
+ * @author      Joakim Nohlgård <joakim.nohlgard@eistec.se>
+ *
+ * @}
+ */
+
+#include <stdint.h>
+#include "phydat.h"
+
+#define ENABLE_DEBUG 0
+#include "debug.h"
+
+int8_t phydat_fit(phydat_t *dat, long value, unsigned int index, int8_t prescale)
+{
+    assert(index < (sizeof(dat->val) / sizeof(dat->val[0])));
+    int8_t ret = prescale;
+    while (prescale > 0) {
+        value /= 10;
+        --prescale;
+    }
+    while (prescale < 0) {
+        value *= 10;
```

Comment:
When would negative `prescale` be used ?
Because `value` could overflow before entering the next loop but I do not really understand if its is a problem or not in practice.

Reply:
Not sure, I just added it for completeness. I am against adding dead code, so I will remove this and change the parameter to uint8_t